### ISSUE TITLE
[BUG] occlum-pal: Failed to run the init process: EINVAL

### ISSUE THREAD

**Issue by AmbitionXiang** [occlum/occlum#937]
# Describe the bug
I meet the same bug as https://github.com/occlum/occlum/issues/364. I also follow https://github.com/occlum/enable_rdfsbase. But with Xeon machine, SGX1, and occlum 0.28.0, which is able to detect the fsgsbase.

# To reproduce
1.  Run the rust demo
2. 
3. 
4. 

# Expected behavior
```
~/demos/rust
/root/demos/rust/occlum_instance initialized as an Occlum instance
Enclave sign-tool: /opt/occlum/sgxsdk-tools/bin/x64/sgx_sign
Enclave sign-key: /opt/occlum/etc/template/Enclave.pem
SGX mode: HW
rm -rf /root/demos/rust/occlum_instance/build
Building new image...
[+] Home dir is /root
[-] Open token file /root/enclave.token error! Will create one.
[+] Saved updated launch token!
[+] Init Enclave Successful 1172526071810!
Generate the SEFS image successfully
Building the initfs...
[+] Home dir is /root
[+] Open token file success! 
[+] Token file valid!
[+] Init Enclave Successful 1241245548546!
Generate the SEFS image successfully
Building libOS...
Signing the enclave...
<EnclaveConfiguration>
    <ProdID>0</ProdID>
    <ISVSVN>0</ISVSVN>
    <StackMaxSize>1048576</StackMaxSize>
    <StackMinSize>1048576</StackMinSize>
    <HeapMaxSize>33554432</HeapMaxSize>
    <HeapMinSize>33554432</HeapMinSize>
    <TCSNum>32</TCSNum>
    <TCSPolicy>1</TCSPolicy>
    <DisableDebug>0</DisableDebug>
    <MiscSelect>0</MiscSelect>
    <MiscMask>0xFFFFFFFF</MiscMask>
    <ReservedMemMaxSize>314572800</ReservedMemMaxSize>
    <ReservedMemMinSize>314572800</ReservedMemMinSize>
    <ReservedMemInitSize>314572800</ReservedMemInitSize>
    <ReservedMemExecutable>1</ReservedMemExecutable>
    <EnableKSS>0</EnableKSS>
    <ISVEXTPRODID_H>0</ISVEXTPRODID_H>
    <ISVEXTPRODID_L>0</ISVEXTPRODID_L>
    <ISVFAMILYID_H>0</ISVFAMILYID_H>
    <ISVFAMILYID_L>0</ISVFAMILYID_L>
</EnclaveConfiguration>
tcs_num 32, tcs_max_num 32, tcs_min_pool 1
The required memory is 386404352B.
The required memory is 0x17081000, 377348 KB.
Succeed.
Built the Occlum image and enclave successfully
[ERROR] occlum-pal: The init process exit with code: 139 (line 62, file src/pal_api.c)
[ERROR] occlum-pal: Failed to run the init process: EINVAL (line 150, file src/pal_api.c)
```

# Logs
```
cd demos/hello_c
make
mkdir occlum_workspace && cd occlum_workspace
occlum init && rm -rf image
copy_bom -f ../hello.yaml --root image --include-dir /opt/occlum/etc/template
occlum build
OCCLUM_LOG_LEVEL=trace occlum run /bin/hello_world
```

The results:
```
[2022-07-01T09:55:38.337Z][DEBUG][T0][#0] lookup_inode: cwd: "/", path: "/etc/resolv.conf"
[2022-07-01T09:55:38.338Z][DEBUG][T0][#0] lookup_inode: cwd: "/", path: "/etc"
[2022-07-01T09:55:38.338Z][DEBUG][T0][#0] real_path: cwd: "/", path: "/etc/resolv.conf"
[2022-07-01T09:55:38.338Z][DEBUG][T0][#0] lookup_inode: cwd: "/", path: "/etc"
[2022-07-01T09:55:38.339Z][DEBUG][T0][#0] convert_to_abs_path: cwd: "/", path: "/etc/resolv.conf"
[2022-07-01T09:55:38.339Z][DEBUG][T0][#0] lookup_inode: cwd: "/", path: "/etc/hostname"
[2022-07-01T09:55:38.339Z][DEBUG][T0][#0] lookup_inode: cwd: "/", path: "/etc"
[2022-07-01T09:55:38.339Z][DEBUG][T0][#0] real_path: cwd: "/", path: "/etc/hostname"
[2022-07-01T09:55:38.339Z][DEBUG][T0][#0] lookup_inode: cwd: "/", path: "/etc"
[2022-07-01T09:55:38.339Z][DEBUG][T0][#0] convert_to_abs_path: cwd: "/", path: "/etc/hostname"
[2022-07-01T09:55:38.339Z][DEBUG][T0][#0] lookup_inode: cwd: "/", path: "/etc/hosts"
[2022-07-01T09:55:38.339Z][DEBUG][T0][#0] convert_to_abs_path: cwd: "/", path: "/etc/hosts"
[2022-07-01T09:55:38.339Z][TRACE][T0][#0] env_checked from env untrusted: []
[2022-07-01T09:55:38.339Z][TRACE][T0][#0] env_merged = ["OCCLUM=yes"]  (default env and untrusted env)
[2022-07-01T09:55:38.339Z][DEBUG][T0][#0] lookup_inode: cwd: "/", path: "/bin/init"
[2022-07-01T09:55:38.340Z][TRACE][T0][#0] PT_INTERP offset = 568, count = 25
[2022-07-01T09:55:38.340Z][TRACE][T0][#0] interpreter = Some("/lib/ld-musl-x86_64.so.1")
[2022-07-01T09:55:38.340Z][TRACE][T0][#0] ldso_path = "/lib/ld-musl-x86_64.so.1"
[2022-07-01T09:55:38.340Z][DEBUG][T0][#0] lookup_inode: cwd: "/", path: "/lib/ld-musl-x86_64.so.1"
[2022-07-01T09:55:38.340Z][TRACE][T0][#0] interpreter = None
[2022-07-01T09:55:38.340Z][DEBUG][T0][#0] allocated rsrv addr is 0x7f7746fed000, len is 0x12c00000
[2022-07-01T09:55:38.340Z][TRACE][T0][#0] find free range, free list = [VMRange { start: 0x7f7746fed000, end: 0x7f7759bed000, size: 0x12c00000 }]
[2022-07-01T09:55:38.340Z][TRACE][T0][#0] after find free range, free list = [VMRange { start: 0x7f7746fed000, end: 0x7f7747000000, size: 0x13000 }, VMRange { start: 0x7f7747400000, end: 0x7f7759bed000, size: 0x127ed000 }]
[2022-07-01T09:55:38.340Z][TRACE][T0][#0] allocate a new single vma chunk: Single VMA chunk: SgxMutex { data: VMArea { range: VMRange { start: 0x7f7747000000, end: 0x7f7747400000, size: 0x400000 }, perms: READ | WRITE | EXEC | DEFAULT | ALL, writeback_file: None, pid: 0 }, poisoned: false, .. }
[2022-07-01T09:55:38.341Z][TRACE][T0][#0] elf range = VMRange { start: 0x7f7747000000, end: 0x7f7747400000, size: 0x400000 }
[2022-07-01T09:55:38.341Z][TRACE][T0][#0] find free range, free list = [VMRange { start: 0x7f7746fed000, end: 0x7f7747000000, size: 0x13000 }, VMRange { start: 0x7f7747400000, end: 0x7f7759bed000, size: 0x127ed000 }]
[2022-07-01T09:55:38.341Z][TRACE][T0][#0] after find free range, free list = [VMRange { start: 0x7f7746fed000, end: 0x7f7747000000, size: 0x13000 }, VMRange { start: 0x7f7747800000, end: 0x7f7759bed000, size: 0x123ed000 }]
[2022-07-01T09:55:38.341Z][TRACE][T0][#0] allocate a new single vma chunk: Single VMA chunk: SgxMutex { data: VMArea { range: VMRange { start: 0x7f7747400000, end: 0x7f7747800000, size: 0x400000 }, perms: READ | WRITE | EXEC | DEFAULT | ALL, writeback_file: None, pid: 0 }, poisoned: false, .. }
[2022-07-01T09:55:38.343Z][TRACE][T0][#0] elf range = VMRange { start: 0x7f7747400000, end: 0x7f7747800000, size: 0x400000 }
[2022-07-01T09:55:38.343Z][TRACE][T0][#0] find free range, free list = [VMRange { start: 0x7f7746fed000, end: 0x7f7747000000, size: 0x13000 }, VMRange { start: 0x7f7747800000, end: 0x7f7759bed000, size: 0x123ed000 }]
[2022-07-01T09:55:38.343Z][TRACE][T0][#0] after find free range, free list = [VMRange { start: 0x7f7746fed000, end: 0x7f7747000000, size: 0x13000 }, VMRange { start: 0x7f7749800000, end: 0x7f7759bed000, size: 0x103ed000 }]
[2022-07-01T09:55:38.343Z][TRACE][T0][#0] allocate a new single vma chunk: Single VMA chunk: SgxMutex { data: VMArea { range: VMRange { start: 0x7f7747800000, end: 0x7f7749800000, size: 0x2000000 }, perms: READ | WRITE | DEFAULT, writeback_file: None, pid: 0 }, poisoned: false, .. }
[2022-07-01T09:55:38.343Z][TRACE][T0][#0] heap range = VMRange { start: 0x7f7747800000, end: 0x7f7749800000, size: 0x2000000 }
[2022-07-01T09:55:38.343Z][TRACE][T0][#0] find free range, free list = [VMRange { start: 0x7f7746fed000, end: 0x7f7747000000, size: 0x13000 }, VMRange { start: 0x7f7749800000, end: 0x7f7759bed000, size: 0x103ed000 }]
[2022-07-01T09:55:38.343Z][TRACE][T0][#0] after find free range, free list = [VMRange { start: 0x7f7746fed000, end: 0x7f7747000000, size: 0x13000 }, VMRange { start: 0x7f7749c00000, end: 0x7f7759bed000, size: 0xffed000 }]
[2022-07-01T09:55:38.343Z][TRACE][T0][#0] allocate a new single vma chunk: Single VMA chunk: SgxMutex { data: VMArea { range: VMRange { start: 0x7f7749800000, end: 0x7f7749c00000, size: 0x400000 }, perms: READ | WRITE | DEFAULT, writeback_file: None, pid: 0 }, poisoned: false, .. }
[2022-07-01T09:55:38.343Z][TRACE][T0][#0] stack range = VMRange { start: 0x7f7749800000, end: 0x7f7749c00000, size: 0x400000 }
[2022-07-01T09:55:38.343Z][TRACE][T0][#0] new process sigmask = SigSet { [] }
[2022-07-01T09:55:38.343Z][TRACE][T0][#0] new process sig_dispositions = SigDispositions (only none-default is shown) {}
[2022-07-01T09:55:38.343Z][DEBUG][T0][#0] process group:ProcessGrp { inner: RwLock { inner: RwLockInner { status: AtomicRwLockStatus(0), rw_waiters: 0 }, data: PgrpInner { pgid: 1, process_group: {1: Process { pid: 1, exec_path: "/bin/init", ppid: Some(0), pgid: 1, inner: ProcessInner::Live { status: Running, child_pids: [], thread_tids: [1] } }}, leader_process: Some(Process { pid: 1, exec_path: "/bin/init", ppid: Some(0), pgid: 1, inner: ProcessInner::Live { status: Running, child_pids: [], thread_tids: [1] } }) } } }
[2022-07-01T09:55:38.343Z][DEBUG][T0][#0] non idle process all pgrp: [ProcessGrp { inner: RwLock { inner: RwLockInner { status: AtomicRwLockStatus(0), rw_waiters: 0 }, data: PgrpInner { pgid: 1, process_group: {1: Process { pid: 1, exec_path: "/bin/init", ppid: Some(0), pgid: 1, inner: ProcessInner::Live { status: Running, child_pids: [], thread_tids: [1] } }}, leader_process: Some(Process { pid: 1, exec_path: "/bin/init", ppid: Some(0), pgid: 1, inner: ProcessInner::Live { status: Running, child_pids: [], thread_tids: [1] } }) } } }, ProcessGrp { inner: RwLock { inner: RwLockInner { status: AtomicRwLockStatus(0), rw_waiters: 0 }, data: PgrpInner { pgid: 0, process_group: {}, leader_process: Some(Process { pid: 0, exec_path: "", ppid: None, pgid: 0, inner: ProcessInner::Live { status: Running, child_pids: [1], thread_tids: [0] } }) } } }]
[2022-07-01T09:55:38.344Z][ INFO][T0][#0] Process created: elf = /bin/init, pid = 1
[2022-07-01T09:55:38.344Z][TRACE][T1][#1][HandleException] Syscall { num = HandleException, info = 0x7f7742528ed0, fpregs = 0x7f7742528c70, context = 0xa3a5301e6cdb53c1 }
[2022-07-01T09:55:38.344Z][DEBUG][T1][#1][HandleException] Handle signal: signal: FaultSignal { num: SigNum (#11 = SIGSEGV), code: 2, addr: Some(140150273576960) }, action: Dfl
[2022-07-01T09:55:38.344Z][TRACE][T1][#1][HandleException] Retval = 0x0
[2022-07-01T09:55:38.344Z][TRACE][T1][#1][HandleException] munmap_chunk range = VMRange { start: 0x7f7747400000, end: 0x7f7747800000, size: 0x400000 }, munmap_range = None
[2022-07-01T09:55:38.344Z][TRACE][T1][#1][HandleException] munmap_chunk range = VMRange { start: 0x7f7747800000, end: 0x7f7749800000, size: 0x2000000 }, munmap_range = None
[2022-07-01T09:55:38.347Z][TRACE][T1][#1][HandleException] munmap_chunk range = VMRange { start: 0x7f7749800000, end: 0x7f7749c00000, size: 0x400000 }, munmap_range = None
[2022-07-01T09:55:38.347Z][TRACE][T1][#1][HandleException] munmap_chunk range = VMRange { start: 0x7f7747000000, end: 0x7f7747400000, size: 0x400000 }, munmap_range = None
[2022-07-01T09:55:38.347Z][DEBUG][T1][#1][HandleException] futex_wake_bitset addr: 0x7f774098d260, max_count: 1, bitset: 0xffffffff
[2022-07-01T09:55:38.349Z][ INFO][T1][#1][HandleException] Thread killed: tid = 1, signum = SigNum (#11 = SIGSEGV)
[2022-07-01T09:55:38.349Z][ INFO][T0][#1][HandleException] Process VM dropped
[2022-07-01T09:55:38.349Z][ INFO][T0][#1][HandleException] Idle process reaps zombie children pid = []
[ERROR] occlum-pal: The init process exit with code: 139 (line 62, file src/pal_api.c)
[ERROR] occlum-pal: Failed to run the init process: EINVAL (line 150, file src/pal_api.c)
[2022-07-01T09:55:38.349Z][ INFO][T0][#0] free user space VM: VMRange { start: 0x7f7746fed000, end: 0x7f7759bed000, size: 0x12c00000 }
```

And logs in occlum gdb:
```
root@e4d68fdfe4f0:~/demos/hello_c/occlum_workspace# occlum gdb /bin/hello_world
GNU gdb (Ubuntu 9.2-0ubuntu1~20.04.1) 9.2
Copyright (C) 2020 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
Type "show copying" and "show warranty" for details.
This GDB was configured as "x86_64-linux-gnu".
Type "show configuration" for configuration details.
For bug reporting instructions, please see:
<http://www.gnu.org/software/gdb/bugs/>.
Find the GDB manual and other documentation resources online at:
    <http://www.gnu.org/software/gdb/documentation/>.

For help, type "help".
Type "apropos word" to search for commands related to "word"...
Source directories searched: /opt/occlum/sgxsdk-tools/lib64/gdb-sgx-plugin:$cdir:$cwd
Setting environment variable "LD_PRELOAD" to null value.
Reading symbols from /root/demos/hello_c/occlum_workspace/build/bin/occlum-run...
(No debugging symbols found in /root/demos/hello_c/occlum_workspace/build/bin/occlum-run)
Loading libc++ pretty-printers.
(gdb) set environment OCCLUM_LOG_LEVEL=trace
(gdb) run 
Starting program: /root/demos/hello_c/occlum_workspace/build/bin/occlum-run /bin/hello_world
warning: Error disabling address space randomization: Operation not permitted
detect urts is loaded, initializing
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".
detect urts is loaded, initializing
add-symbol-file '/root/demos/hello_c/occlum_workspace/build/lib/libocclum-libos.signed.so' 0x7fe640047040 -s .interp 0x7fe6400002e0  -s .note.gnu.build-id 0x7fe6400002fc  -s .gnu.hash 0x7fe640000320  -s .dynsym 0x7fe640000358  -s .dynstr 0x7fe640000400  -s .gnu.version 0x7fe64000047a  -s .gnu.version_d 0x7fe640000488  -s .rela.dyn 0x7fe6400004c0  -s .plt 0x7fe640047000  -s .plt.got 0x7fe640047010  -s .nipx 0x7fe6402d0710  -s .rodata 0x7fe6402d2000  -s .niprod 0x7fe640334000  -s .builtin_config 0x7fe640334900  -s .eh_frame_hdr 0x7fe640334930  -s .eh_frame 0x7fe640340450  -s .gcc_except_table 0x7fe640386ce4  -s .tbss 0x7fe64039a780  -s .init_array 0x7fe64039a780  -s .fini_array 0x7fe64039a798  -s .data.rel.ro 0x7fe64039a800  -s .dynamic 0x7fe6403cc820  -s .got 0x7fe6403cc9c0  -s .data 0x7fe6403ce000  -s .nipd 0x7fe6403d1340  -s .bss 0x7fe6403d1380 
[New Thread 0x7fe669cbd700 (LWP 565)]
[2022-07-01T10:15:19.223Z][DEBUG][T0][#0] lookup_inode: cwd: "/", path: "/etc/resolv.conf"
[2022-07-01T10:15:19.237Z][DEBUG][T0][#0] convert_to_abs_path: cwd: "/", path: "/etc/resolv.conf"
[2022-07-01T10:15:19.239Z][DEBUG][T0][#0] lookup_inode: cwd: "/", path: "/etc/hostname"
[2022-07-01T10:15:19.241Z][DEBUG][T0][#0] convert_to_abs_path: cwd: "/", path: "/etc/hostname"
[2022-07-01T10:15:19.243Z][DEBUG][T0][#0] lookup_inode: cwd: "/", path: "/etc/hosts"
[2022-07-01T10:15:19.246Z][DEBUG][T0][#0] convert_to_abs_path: cwd: "/", path: "/etc/hosts"
[New Thread 0x7fe6694bc700 (LWP 566)]
[2022-07-01T10:15:19.248Z][TRACE][T0][#0] env_checked from env untrusted: []
[2022-07-01T10:15:19.248Z][TRACE][T0][#0] env_merged = ["OCCLUM=yes"]  (default env and untrusted env)
[2022-07-01T10:15:19.249Z][DEBUG][T0][#0] lookup_inode: cwd: "/", path: "/bin/init"
[2022-07-01T10:15:19.254Z][TRACE][T0][#0] PT_INTERP offset = 568, count = 25
[2022-07-01T10:15:19.254Z][TRACE][T0][#0] interpreter = Some("/lib/ld-musl-x86_64.so.1")
[2022-07-01T10:15:19.255Z][TRACE][T0][#0] ldso_path = "/lib/ld-musl-x86_64.so.1"
[2022-07-01T10:15:19.255Z][DEBUG][T0][#0] lookup_inode: cwd: "/", path: "/lib/ld-musl-x86_64.so.1"
[2022-07-01T10:15:19.259Z][TRACE][T0][#0] interpreter = None
[2022-07-01T10:15:19.259Z][DEBUG][T0][#0] allocated rsrv addr is 0x7fe646fed000, len is 0x12c00000
[2022-07-01T10:15:19.260Z][TRACE][T0][#0] find free range, free list = [VMRange { start: 0x7fe646fed000, end: 0x7fe659bed000, size: 0x12c00000 }]
[2022-07-01T10:15:19.260Z][TRACE][T0][#0] after find free range, free list = [VMRange { start: 0x7fe646fed000, end: 0x7fe647000000, size: 0x13000 }, VMRange { start: 0x7fe647400000, end: 0x7fe659bed000, size: 0x127ed000 }]
[2022-07-01T10:15:19.261Z][TRACE][T0][#0] allocate a new single vma chunk: Single VMA chunk: SgxMutex { data: VMArea { range: VMRange { start: 0x7fe647000000, end: 0x7fe647400000, size: 0x400000 }, perms: READ | WRITE | EXEC | DEFAULT | ALL, writeback_file: None, pid: 0 }, poisoned: false, .. }
[2022-07-01T10:15:19.301Z][TRACE][T0][#0] elf range = VMRange { start: 0x7fe647000000, end: 0x7fe647400000, size: 0x400000 }
[2022-07-01T10:15:19.301Z][TRACE][T0][#0] find free range, free list = [VMRange { start: 0x7fe646fed000, end: 0x7fe647000000, size: 0x13000 }, VMRange { start: 0x7fe647400000, end: 0x7fe659bed000, size: 0x127ed000 }]
[2022-07-01T10:15:19.302Z][TRACE][T0][#0] after find free range, free list = [VMRange { start: 0x7fe646fed000, end: 0x7fe647000000, size: 0x13000 }, VMRange { start: 0x7fe647800000, end: 0x7fe659bed000, size: 0x123ed000 }]
[2022-07-01T10:15:19.302Z][TRACE][T0][#0] allocate a new single vma chunk: Single VMA chunk: SgxMutex { data: VMArea { range: VMRange { start: 0x7fe647400000, end: 0x7fe647800000, size: 0x400000 }, perms: READ | WRITE | EXEC | DEFAULT | ALL, writeback_file: None, pid: 0 }, poisoned: false, .. }
[2022-07-01T10:15:19.365Z][TRACE][T0][#0] elf range = VMRange { start: 0x7fe647400000, end: 0x7fe647800000, size: 0x400000 }
[2022-07-01T10:15:19.366Z][TRACE][T0][#0] find free range, free list = [VMRange { start: 0x7fe646fed000, end: 0x7fe647000000, size: 0x13000 }, VMRange { start: 0x7fe647800000, end: 0x7fe659bed000, size: 0x123ed000 }]
[2022-07-01T10:15:19.366Z][TRACE][T0][#0] after find free range, free list = [VMRange { start: 0x7fe646fed000, end: 0x7fe647000000, size: 0x13000 }, VMRange { start: 0x7fe649800000, end: 0x7fe659bed000, size: 0x103ed000 }]
[2022-07-01T10:15:19.366Z][TRACE][T0][#0] allocate a new single vma chunk: Single VMA chunk: SgxMutex { data: VMArea { range: VMRange { start: 0x7fe647800000, end: 0x7fe649800000, size: 0x2000000 }, perms: READ | WRITE | DEFAULT, writeback_file: None, pid: 0 }, poisoned: false, .. }
[2022-07-01T10:15:19.367Z][TRACE][T0][#0] heap range = VMRange { start: 0x7fe647800000, end: 0x7fe649800000, size: 0x2000000 }
[2022-07-01T10:15:19.367Z][TRACE][T0][#0] find free range, free list = [VMRange { start: 0x7fe646fed000, end: 0x7fe647000000, size: 0x13000 }, VMRange { start: 0x7fe649800000, end: 0x7fe659bed000, size: 0x103ed000 }]
[2022-07-01T10:15:19.368Z][TRACE][T0][#0] after find free range, free list = [VMRange { start: 0x7fe646fed000, end: 0x7fe647000000, size: 0x13000 }, VMRange { start: 0x7fe649c00000, end: 0x7fe659bed000, size: 0xffed000 }]
[2022-07-01T10:15:19.368Z][TRACE][T0][#0] allocate a new single vma chunk: Single VMA chunk: SgxMutex { data: VMArea { range: VMRange { start: 0x7fe649800000, end: 0x7fe649c00000, size: 0x400000 }, perms: READ | WRITE | DEFAULT, writeback_file: None, pid: 0 }, poisoned: false, .. }
[2022-07-01T10:15:19.369Z][TRACE][T0][#0] stack range = VMRange { start: 0x7fe649800000, end: 0x7fe649c00000, size: 0x400000 }
add-symbol-file 'image/lib/ld-musl-x86_64.so.1' 0x7fe6474142f0 -s .hash 0x7fe647400190  -s .gnu.hash 0x7fe647402c48  -s .dynsym 0x7fe647405f08  -s .dynstr 0x7fe64740fe80  -s .rela.dyn 0x7fe647413bb8  -s .rela.plt 0x7fe647414218  -s .plt 0x7fe647414290  -s .rodata 0x7fe64745d700  -s .eh_frame_hdr 0x7fe6474912c8  -s .eh_frame 0x7fe6474912f8  -s .data.rel.ro 0x7fe647691c40  -s .dynamic 0x7fe647691e20  -s .got 0x7fe647691f70  -s .got.plt 0x7fe647692000  -s .data 0x7fe647692040  -s .bss 0x7fe6476924a0 
Function "_dl_map_object" not defined.
readelf: Error: 'image/bin/init': No such file
[2022-07-01T10:15:19.381Z][TRACE][T0][#0] new process sigmask = SigSet { [] }
[2022-07-01T10:15:19.381Z][TRACE][T0][#0] new process sig_dispositions = SigDispositions (only none-default is shown) {}
[2022-07-01T10:15:19.382Z][DEBUG][T0][#0] process group:ProcessGrp { inner: RwLock { inner: RwLockInner { status: AtomicRwLockStatus(0), rw_waiters: 0 }, data: PgrpInner { pgid: 1, process_group: {1: Process { pid: 1, exec_path: "/bin/init", ppid: Some(0), pgid: 1, inner: ProcessInner::Live { status: Running, child_pids: [], thread_tids: [1] } }}, leader_process: Some(Process { pid: 1, exec_path: "/bin/init", ppid: Some(0), pgid: 1, inner: ProcessInner::Live { status: Running, child_pids: [], thread_tids: [1] } }) } } }
[2022-07-01T10:15:19.383Z][DEBUG][T0][#0] non idle process all pgrp: [ProcessGrp { inner: RwLock { inner: RwLockInner { status: AtomicRwLockStatus(0), rw_waiters: 0 }, data: PgrpInner { pgid: 1, process_group: {1: Process { pid: 1, exec_path: "/bin/init", ppid: Some(0), pgid: 1, inner: ProcessInner::Live { status: Running, child_pids: [], thread_tids: [1] } }}, leader_process: Some(Process { pid: 1, exec_path: "/bin/init", ppid: Some(0), pgid: 1, inner: ProcessInner::Live { status: Running, child_pids: [], thread_tids: [1] } }) } } }, ProcessGrp { inner: RwLock { inner: RwLockInner { status: AtomicRwLockStatus(0), rw_waiters: 0 }, data: PgrpInner { pgid: 0, process_group: {}, leader_process: Some(Process { pid: 0, exec_path: "", ppid: None, pgid: 0, inner: ProcessInner::Live { status: Running, child_pids: [1], thread_tids: [0] } }) } } }]
[2022-07-01T10:15:19.383Z][ INFO][T0][#0] Process created: elf = /bin/init, pid = 1

Thread 1 "occlum-run" received signal SIGSEGV, Segmentation fault.
0x00007fe647459c1d in _dlstart ()
(gdb) 
```

# Environment
- HW:  Intel(R) Xeon(R) Gold 5317 CPU @ 3.00GHz, SGX1 
- OS: Ubuntu18.04, kernel 5.4.202, patched to enable fsgsbase by following https://gramine.readthedocs.io/en/stable/devel/building.html#advanced-installing-linux-kernel-with-fsgsbase-patches
- Occlum version: 0.28.0
- Intel SGX SDK 2.14 with out-of-tree driver

# Additional context
On another machine, Intel(R) Core(TM) i7-9700K CPU @ 3.60GHz, ubuntu 18.04 with kernel 5.4.169, also patched by following https://gramine.readthedocs.io/en/stable/devel/building.html#advanced-installing-linux-kernel-with-fsgsbase-patches, the error does not occur.

# Possible solution/Implementation
Not obligatory, but suggest a fix/reason for the bug.

**Comment by volcano0dr** [comment]
`Occlum 0.28.0 `has not been released yet. In addition `0.28.0` requires the use of `Intel SGX SDK 2.16`
Please try occlum/occlum:0.27.3-ubuntu20.04.

**Comment by AmbitionXiang** [comment]
Actually, when I tried 0.27.0, I have met this error after solving https://github.com/occlum/occlum/issues/933. The error report is the same as https://github.com/occlum/occlum/issues/364

**Comment by guzongmin** [comment]
@AmbitionXiang In some versions of Kernel, the rdfsbase driver is blocked. So please update your kernel to a newer version or regression to an older version. I suggest you update to 5.11 or a later one.

**Comment by AmbitionXiang** [comment]
Hope it may be helpful for you to fix the bug...
I replaced the out-of-tree driver with DCAP driver v1.41, and then the problem was solved. I still used Occlum 0.28.0 and SGX SDK 2.14.
Moreover, the patch https://github.com/occlum/enable_rdfsbase seems useless, while I patched kernel 5.4.0 successfully by following https://gramine.readthedocs.io/en/stable/devel/building.html#advanced-installing-linux-kernel-with-fsgsbase-patches.

**Comment by guzongmin** [comment]
@AmbitionXiang Happy to know you make it work. The fsgsbase support and SGX driver are merged into the kernel 5.11 and later versions. So using the officially supported kernel may be a better solution. There is a security issue (only impacts the applications running outside the enclave) with the simple kernel patch or driver. For your reference, [https://lwn.net/Articles/821723/](https://lwn.net/Articles/821723/). So the better solution is to use a new kernel.

**Comment by AmbitionXiang** [comment]
Thank you. Closing the issue...
